lib/Sema/TypeCheckMacros.cpp

```diff
@@ -103,20 +106,50 @@ Expr *swift::expandMacroExpr(
   if (!sourceFile)
     return nullptr;
 
-  auto astGenSourceFile = sourceFile->exportedSourceFile;
-  if (!astGenSourceFile)
-    return nullptr;
-
   // Evaluate the macro.
-  const char *evaluatedSource;
-  ptrdiff_t evaluatedSourceLength;
-  swift_ASTGen_evaluateMacro(
-      astGenSourceFile, expr->getStartLoc().getOpaquePointerValue(),
-      &evaluatedSource, &evaluatedSourceLength);
-  if (!evaluatedSource)
-    return nullptr;
-
-  SourceManager &sourceMgr = ctx.SourceMgr;
+  StringRef evaluatedSource;
+
+  // Built-in macros go through `MacroSystem` in Swift Syntax linked to this
+  // compiler.
+  if (isBuiltinMacro) {
```

Comment:
This is fine for now, but we should seek to unify them by having a single `Macro` structure that abstracts over builtin vs. loaded from a plugin so the rest of the compiler doesn't care.

Reply:
I think handling of both of these can be cleaned up a bit, so don't fret over it in this PR.